10-habits/5_integration/integration/integration_test.go

```diff
@@ -0,0 +1,113 @@
+package integration
+
+import (
+	"context"
+	"net"
+	"testing"
+
+	"learngo-pockets/habits/internal/repository"
+	"learngo-pockets/habits/internal/server"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+
+	"learngo-pockets/habits/api"
+)
+
+func TestIntegration(t *testing.T) {
+	// run server
+	grpcServ := newServer(t)
```

Comment:
Rather than creating a server using grpc use the `server/` package directly, otherwise you end up testing different code -- or duplicating code in your tests.

Reply:
We decided to go differently and did a bit of refactoring in the Listen method: https://github.com/alienorlatour/tiny-go-projects/pull/95/files